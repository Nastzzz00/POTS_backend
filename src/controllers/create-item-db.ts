import { IDBModel } from '../commons/types';
import { createUCCreateItem } from '../core/';

const createCreateItemDB = (db: IDBModel<any>) => async i => {
  const createItem = createUCCreateItem();
  const newItem = createItem(i);
  return db.insert(newItem);
};

export { createCreateItemDB };
