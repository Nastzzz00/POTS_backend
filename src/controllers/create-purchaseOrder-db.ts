import { IDBModel } from '../commons/types';
import { createUCPurchaseOrder } from '../core/';

const createCreatePurchaseOrderDB = (db: IDBModel<any>) => async po => {
	const createPurchaseOrder = createUCPurchaseOrder();
	const newPurchaseOrder = createPurchaseOrder(po);
	return db.insert(newPurchaseOrder);
};

export { createCreatePurchaseOrderDB };
