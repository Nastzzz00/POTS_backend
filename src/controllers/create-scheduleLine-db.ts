import { IDBModel } from '../commons/types';
import { createUCCreateScheduleLine } from '../core/';

const createCreateScheduleLineDB = (db: IDBModel<any>) => async sl => {
	const createScheduleLine = createUCCreateScheduleLine();
	const newScheduleLine = createScheduleLine(sl);
	return db.insert(newScheduleLine);
};

export { createCreateScheduleLineDB };
