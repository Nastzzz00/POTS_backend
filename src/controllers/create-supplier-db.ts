import { IDBModel } from '../commons/types';
import { createUCCreateSupplier } from '../core/';

const createCreateSupplierDB = (db: IDBModel<any>) => async u => {
	const createSupplier = createUCCreateSupplier();
	const newSupplier = createSupplier(u);
	return db.insert(newSupplier);
};

export { createCreateSupplierDB };
