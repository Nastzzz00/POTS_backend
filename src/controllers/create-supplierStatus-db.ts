import { IDBModel } from '../commons/types';
import { createUCCreateSupplierStatus } from '../core/';

const createCreateSupplierStatusDB = (db: IDBModel<any>) => async u => {
	const createSupplierStatus = createUCCreateSupplierStatus();
	const newSupplierStatus = createSupplierStatus(u);
	return db.insert(newSupplierStatus);
};

export { createCreateSupplierStatusDB };
