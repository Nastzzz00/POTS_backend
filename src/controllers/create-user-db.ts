import { IDBModel } from '../commons/types';
import { createUCCreateUser } from '../core/';

const createCreateUserDB = (db: IDBModel<any>) => async u => {
	const createUser = createUCCreateUser();
	const newUser = createUser(u);
	return db.insert(newUser);
};

export { createCreateUserDB };
