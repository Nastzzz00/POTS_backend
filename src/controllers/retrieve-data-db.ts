import { IDBModel } from '../commons/types';

const getAllDataDB = (db: IDBModel<any>) => async () => {
	return db.getAll();
};
const getByIDDB = (db: IDBModel<any>) => async id => {
	return db.getById(id);
};

const getAllBySupplierStatusDB = (db: IDBModel<any>) => async id => {
	return db.getAllBySupplierStatus(id);
};

const getAllByItemDB = (db: IDBModel<any>) => async id => {
	return db.getAllByItem(id);
};

const getAllByScheduleLineDB = (db: IDBModel<any>) => async id => {
	return db.getAllByScheduleLine(id);
};

export { getAllDataDB, getByIDDB, getAllByItemDB, getAllBySupplierStatusDB, getAllByScheduleLineDB };
