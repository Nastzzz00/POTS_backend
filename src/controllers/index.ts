import {
  getAllDataDB,
  getByIDDB,
  getAllByItemDB,
  getAllBySupplierStatusDB,
  getAllByScheduleLineDB,
} from './retrieve-data-db';
import { DeleteRecordByIDDB } from './delete-data-db';

//User
import { createCreateUserDB } from './create-user-db';
import { updateUserByIDDB } from './update-user-db';

//Supplier
import { createCreateSupplierDB } from './create-supplier-db';
import { updateSupplierByIDDB } from './update-supplier-db';

//Supplier Status
import { createCreateSupplierStatusDB } from './create-supplierStatus-db';
import { updateSupplierStatusByIDDB } from './update-supplierStatus-db';
import { updateSupplierStatusItemByIDDB } from './update-supplierStatusItem-db';

//Item
import { createCreateItemDB } from './create-item-db';
import { updateItemByIDDB } from './update-item-db';

//Purchase Order
import { createCreatePurchaseOrderDB } from './create-purchaseOrder-db';
import { updatePurchaseOrderByIDDB } from './update-purchaseOrder-db';
import { updateAdminStatusByIDDB } from './update-adminStatus-db';

//Schedule Line
import { createCreateScheduleLineDB } from './create-scheduleLine-db';
import { updateScheduleLineByIDDB } from './update-scheduleLine-db';

export {
  getAllDataDB,
  getByIDDB,
  DeleteRecordByIDDB,
  getAllByItemDB,
  getAllBySupplierStatusDB,
  getAllByScheduleLineDB,
  //User exports
  createCreateUserDB,
  updateUserByIDDB,
  //Supplier exports
  createCreateSupplierDB,
  updateSupplierByIDDB,
  //SupplierStatus exports
  createCreateSupplierStatusDB,
  updateSupplierStatusByIDDB,
  //Item
  createCreateItemDB,
  updateItemByIDDB,
  updateSupplierStatusItemByIDDB,
  //Purchase Order
  createCreatePurchaseOrderDB,
  updatePurchaseOrderByIDDB,
  updateAdminStatusByIDDB,
  //Schedule Line
  createCreateScheduleLineDB,
  updateScheduleLineByIDDB,
};
