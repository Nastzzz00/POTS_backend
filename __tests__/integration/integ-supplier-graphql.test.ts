import { createTestClient } from 'apollo-server-testing';
import gql from 'graphql-tag';
import mockData from '../mocks/mock-data';
import * as controllers from '../../src/controllers';
import { constructTestServer } from '../__utils';

const {
  //Generic
  getAllDataDB,
  getByIDDB,
  DeleteRecordByIDDB,
  createCreateSupplierDB,
  updateSupplierByIDDB,
} = controllers;

const supplierMock = {
  insert: jest.fn(async input => {
    return { id: '1', ...input, address: 'A2' };
  }),
  getAll: jest.fn(async () => {
    return mockData.suppliers;
  }),
  getById: jest.fn(async id => {
    const filterData = cust => {
      if (cust.id === id) {
        return cust;
      }
    };

    const res = mockData.suppliers.filter(filterData);
    return res[0] || null;
  }),

  deleteById: async id => {
    const filterData = cust => {
      if (cust.id === id) {
        return cust;
      }
    };

    const res = mockData.suppliers.filter(filterData);
    return res[0] || null;
  },

  updateById: jest.fn(async input => {
    return { ...input };
  }),
  getAllBySupplierStatus: async id => {},
  getAllByItem: async id => {},
  getAllByScheduleLine: async id => {},
  updateSupplierStatusItemById: async id => {},
  updateAdminStatusPurchaseOrderById: async id => {},
};

const addressMock = {
  insert: jest.fn(async input => {
    return { id: 'A2', ...input };
  }),
  getAll: jest.fn(async () => {
    return mockData.address;
  }),
  getById: jest.fn(async id => {
    const filterData = data => {
      if (data.id === id) {
        return data;
      }
    };
    const res = mockData.address.filter(filterData);
    return res[0] || null;
  }),
  deleteById: async id => {},
  updateById: async id => {},
  getAllBySupplierStatus: async id => {},
  getAllByItem: async id => {},
  getAllByScheduleLine: async id => {},
  updateSupplierStatusItemById: async id => {},
  updateAdminStatusPurchaseOrderById: async id => {},
};

const { server }: any = constructTestServer({
  context: {
    createSupplier: createCreateSupplierDB(supplierMock),
    getAllSuppliers: getAllDataDB(supplierMock),
    getSupplierById: getByIDDB(supplierMock),
    deleteSupplierById: DeleteRecordByIDDB(supplierMock),
    updateSupplierById: updateSupplierByIDDB(supplierMock),
    getAddressById: getByIDDB(addressMock),
  },
});

describe('Tests', () => {
  //Queries
  it('should fetch all suppliers', async () => {
    const SUPPLIER_ALL = gql`
      query {
        allSuppliers {
          id
          supplierNo
          supplierName
          address {
            id
            building_name
            street
            city
            state
            zip_code
          }
          contactPerson
          contactNumber
          tin
        }
      }
    `;

    const { query } = createTestClient(server);
    const res = await query({ query: SUPPLIER_ALL });

    expect(res).toMatchSnapshot();
  });

  it('should fetch one supplier', async () => {
    const SINGLE_SUPPLIER = gql`
      query supp($id: ID!) {
        supplier(id: $id) {
          id
          supplierNo
          supplierName
          address {
            id
            building_name
            street
            city
            state
            zip_code
          }
          contactPerson
          contactNumber
          tin
        }
      }
    `;

    const { query } = createTestClient(server);
    const res = await query({
      query: SINGLE_SUPPLIER,
      variables: { id: '1' },
    });

    expect(res).toMatchSnapshot();
  });

  it('should error when no supplier', async () => {
    const SINGLE_SUPPLIER = gql`
      query supp($id: ID!) {
        supplier(id: $id) {
          id
          supplier {
            supplierNo
            supplierName
            address {
              id
              building_name
              street
              city
              state
              zip_code
            }
            contactPerson
            contactNumber
            tin
          }
        }
      }
    `;

    const { query } = createTestClient(server);
    const res = await query({
      query: SINGLE_SUPPLIER,
      variables: { id: '' },
    });

    expect(res).toMatchSnapshot();
  });

  //Mutations

  it('create a supplier', async () => {
    const CREATE_SUPPLIER = gql`
      mutation createSupp($supplier: SupplierInput!) {
        createSupplier(supplier: $supplier) {
          id
          supplierNo
          supplierName
          address {
            id
            building_name
            street
            city
            state
            zip_code
          }
          contactPerson
          contactNumber
          tin
        }
      }
    `;

    const { mutate } = createTestClient(server);
    const res = await mutate({
      mutation: CREATE_SUPPLIER,
      variables: {
        supplier: {
          supplierNo: '001',
          supplierName: 'Juan Dela Cruz',
          address: {
            building_name: '002',
            street: 'Elmer',
            city: 'Celadon',
            state: 'Johto',
            zip_code: '123',
          },
          contactPerson: 'Basil Valdez',
          contactNumber: '1234567',
          tin: '12345',
        },
      },
    });

    expect(res.errors).toBeUndefined();
    expect(supplierMock.insert.mock.calls.length).toBe(1);

    expect(res.data).toMatchObject({
      createSupplier: {
        id: '1',
        supplierNo: '001',
        supplierName: 'Juan Dela Cruz',
        address: {
          id: 'A2',
          building_name: '002',
          street: 'Elmer',
          city: 'Celadon',
          state: 'Johto',
          zip_code: '123',
        },
        contactPerson: 'Basil Valdez',
        contactNumber: '1234567',
        tin: '12345',
      },
    });
    expect(res).toMatchSnapshot();
  });

  it('update a supplier', async () => {
    const UPDATE_SUPPLIER = gql`
      mutation supp($supplier: UpdateSupplierInput!) {
        updateSupplier(supplier: $supplier) {
          id
          supplierNo
          supplierName
          contactPerson
          contactNumber
          tin
        }
      }
    `;

    const { mutate } = createTestClient(server);
    const res = await mutate({
      mutation: UPDATE_SUPPLIER,
      variables: {
        supplier: {
          id: '1',
          supplierNo: '001',
          supplierName: 'Juan Dela Cruz',
          contactPerson: 'Basil Valdez',
          contactNumber: '1234567',
          tin: '12345',
        },
      },
    });

    expect(res).toMatchSnapshot();
  });

  it('delete a supplier', async () => {
    const DELETE_SUPPLIER = gql`
      mutation supp($id: ID!) {
        deleteSupplier(id: $id) {
          supplierName
        }
      }
    `;

    const { mutate } = createTestClient(server);
    const res = await mutate({
      mutation: DELETE_SUPPLIER,
      variables: { id: 'S1' },
    });

    expect(res).toMatchSnapshot();
  });
});
