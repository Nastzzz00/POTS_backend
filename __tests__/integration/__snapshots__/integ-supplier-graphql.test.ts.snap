// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests create a supplier 1`] = `
Object {
  "data": Object {
    "createSupplier": Object {
      "address": Object {
        "building_name": "002",
        "city": "Celadon",
        "id": "A2",
        "state": "Johto",
        "street": "Elmer",
        "zip_code": "123",
      },
      "contactNumber": "1234567",
      "contactPerson": "Basil Valdez",
      "id": "1",
      "supplierName": "Juan Dela Cruz",
      "supplierNo": "001",
      "tin": "12345",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Tests delete a supplier 1`] = `
Object {
  "data": Object {
    "deleteSupplier": null,
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Tests should error when no supplier 1`] = `
Object {
  "data": undefined,
  "errors": Array [
    [ValidationError: Cannot query field "supplier" on type "Supplier". Did you mean "supplierNo" or "supplierName"?],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Tests should fetch all suppliers 1`] = `
Object {
  "data": Object {
    "allSuppliers": Array [
      Object {
        "address": Object {
          "building_name": "002",
          "city": "Celadon",
          "id": "A2",
          "state": "Johto",
          "street": "Elmer",
          "zip_code": "123",
        },
        "contactNumber": "1234567",
        "contactPerson": "Basil Valdez",
        "id": "1",
        "supplierName": "Juan Dela Cruz",
        "supplierNo": "001",
        "tin": "12345",
      },
      Object {
        "address": Object {
          "building_name": "002",
          "city": "Celadon",
          "id": "A1",
          "state": "Johto",
          "street": "Elmer",
          "zip_code": "123",
        },
        "contactNumber": "1234567",
        "contactPerson": "Basil Valdez",
        "id": "2",
        "supplierName": "Juan Dela Cruz",
        "supplierNo": "002",
        "tin": "12345",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Tests should fetch one supplier 1`] = `
Object {
  "data": Object {
    "supplier": Object {
      "address": Object {
        "building_name": "002",
        "city": "Celadon",
        "id": "A2",
        "state": "Johto",
        "street": "Elmer",
        "zip_code": "123",
      },
      "contactNumber": "1234567",
      "contactPerson": "Basil Valdez",
      "id": "1",
      "supplierName": "Juan Dela Cruz",
      "supplierNo": "001",
      "tin": "12345",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Tests update a supplier 1`] = `
Object {
  "data": Object {
    "updateSupplier": Object {
      "contactNumber": "1234567",
      "contactPerson": "Basil Valdez",
      "id": "1",
      "supplierName": "Juan Dela Cruz",
      "supplierNo": "001",
      "tin": "12345",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
